scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,530 @@
+			return false
+		}
+	}
+
+	return true
+}
+
+func compareTcb(refM comid.Measurement, evM comid.Measurement) bool {
+	if refM.Val.SVN == nil {
+		log.Errorf("reference doesn't have SVN")
+		return false
+	}
+
+	if evM.Val.SVN == nil {
+		log.Errorf("evidence doesn't have SVN")
+		return false
+	}
+
+	refTcbParts, err := transformSVNtoTCB(*refM.Val.SVN)
+	if err != nil {
+		log.Errorf("could not transform reference SVN to TCB parts: %v", err)
+		return false
+	}
+
+	evTcbParts, err := transformSVNtoTCB(*evM.Val.SVN)
+	if err != nil {
+		log.Errorf("could not transform evidence SVN to TCB parts: %v", err)
+	}
+
+	if evTcbParts.BlSpl < refTcbParts.BlSpl ||
+		evTcbParts.SnpSpl < refTcbParts.SnpSpl ||
+		evTcbParts.TeeSpl < refTcbParts.TeeSpl ||
+		evTcbParts.UcodeSpl < refTcbParts.UcodeSpl {
+		return false
+	}
+
+	return true
+}
+
+// AppraiseEvidence confirms if the claims in the evidence match with the provisioned
+// reference values.
+//
+// Appraisal can confirm if the evidence is genuinely generated by AMD
+// hardware and if SEV-SNP enables memory encryption. As such, set the
+// "Hardware" and "RuntimeOpaque" values in the trustworthiness vector;
+// we can't infer other aspects of the vector from SEV-SNP evidence alone.
+func (o EvidenceHandler) AppraiseEvidence(
+	ec *proto.EvidenceContext,
+	endorsementsStrings []string,
+) (*ear.AttestationResult, error) {
+	var (
+		err         error
+		evidenceMap map[string]interface{}
+	)
+
+	refVal, err := refvalToComidTriple(endorsementsStrings)
+	if err != nil {
+		return nil, err
+	}
+
+	evidence, err := evidenceToComidTriple(ec)
+	if err != nil {
+		return nil, err
+	}
+
+	result := handler.CreateAttestationResult(SchemeName)
+
+	appraisal := result.Submods[SchemeName]
+
+	appraisal.TrustVector.InstanceIdentity = ear.NoClaim
+	appraisal.TrustVector.Executables = ear.NoClaim
+	appraisal.TrustVector.Configuration = ear.NoClaim
+	appraisal.TrustVector.FileSystem = ear.NoClaim
+	appraisal.TrustVector.StorageOpaque = ear.NoClaim
+	appraisal.TrustVector.SourcedData = ear.NoClaim
+	appraisal.TrustVector.Hardware = ear.UnsafeHardwareClaim
+	appraisal.TrustVector.RuntimeOpaque = ear.VisibleMemoryRuntimeClaim
+
+claimsLoop:
+	for _, m := range refVal.Measurements.Values {
+		var (
+			k  uint64
+			em *comid.Measurement
+		)
+
+		k, err = m.Key.GetKeyUint()
+		if err != nil {
+			break
+		}
+
+		// We can skip validating certain claims for the following reasons:
+		// - POLICY ToDo: Do we need to test individual policy features?
+		// - CURRENT_TCB is informational only. It's best handled by policy
+		// - PLATFORM_INFO ToDO: Do we need to test individual platform features?
+		// - REPORT_DATA is a nonce supplied by user for freshness, not applicable for verification
+		// - REPORT_ID is ephemeral, so we can't use it for verification.
+		// - REPORT_ID_MA is also ephemeral, used for migration
+		// - CHIP_ID is unique to an specific attester, but reference values could be used more generally
```

Comment:
Just a question:

Does the CHIP_ID maps to a specific instance of platform??? If Yes, shall we not populate the InstanceID Claims..